lib/manageiq/api/common/user.rb

```diff
@@ -0,0 +1,87 @@
+module ManageIQ
+  module API
+    module Common
+      class User
+        IDENTITY_KEY = 'x-rh-identity'.freeze
+
+        def self.current
+          Thread.current[:attr_current_user]
+        end
+
+        def self.current=(user_header)
+          Thread.current[:attr_current_user] = new(user_header)
+        end
+
+        def self.with_logged_in_user(user_header)
+          self.current = user_header
+          yield self.current
+        ensure
+          Thread.current[:attr_current_user] = nil
+        end
+
+        def initialize(user_header)
+          @user = user_header
+        end
+
+        def username
+          validate_and_return(__method__)
+        end
+
+        def email
+          validate_and_return(__method__)
+        end
+
+        def first_name
+          validate_and_return(__method__)
+        end
+
+        def last_name
+          validate_and_return(__method__)
+        end
+
+        def is_active?
+          validate_and_return(:is_active)
+        end
+
+        def is_org_admin?
+          validate_and_return(:is_org_admin)
+        end
+
+        def is_internal?
+          validate_and_return(:is_internal)
+        end
+
+        def locale
+          validate_and_return(__method__)
+        end
+
+        def tenant
+          validate_and_return(:account_number)
+        end
+
+        private
+
+        def decode
+          if @user.is_a?(ActionDispatch::Http::Headers)
+            val = @user.instance_variable_get(:@req).stringify_keys
+          else
+            val = @user.stringify_keys
+          end
+          JSON.parse(Base64.decode64(val[IDENTITY_KEY]))
+        rescue NoMethodError => e
+          raise ArgumentError, "Not a valid user hash: #{e.message}"
+        end
+
+        def validate_and_return(key)
+          find_key(decode, key)
+        end
+
+        def find_key(hash, key)
+          hash['identity'].has_key?(key.to_s) ? hash['identity'][key.to_s] : hash['identity']['user'][key.to_s]
```

Comment:
To address @mkanoor's earlier concern about the case where the first level `['identity']` and the second level `['identity']['user']` contain the same key, we'd better to separate them clearly.
Consider rename this method name to `find_user_key`, and let `tenant` method not calling into this method but directly checking `hash['identity']['account_number']`.

Reply:
Note the method from more_core_extensions that would be helpful: `has_key_path?`

Reply:
This feels easier to read: `hash.fetch_path('identity', key.to_s) || hash.fetch_path('identity', 'user', key.to_s)`

What do you think?